### PR TITLE
Teleport Plugins Documentation

### PR DESCRIPTION
**Pull request by benarent**
This PR ports over Install docs from the Teleport-Plugin Repository. 

Todo
- [x] Add / port over screenshots
- [x] Test Setup 
- [ ] Record Setup Videos

### CODE REVIEW

**Comment by benarent** [comment]
@stevenGravy Thanks for the recent language changes, can you please make any other change directly to this PR.

File: docs/4.3/enterprise/workflow/ssh_approval_jira_server.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+Go ahead and test it: 
+
+```bash
+tsh login --request-roles=admin
+```
+
+That should create a new permission request on Teleport (you can test if it did with `tctl request ls` ), and you should see a new task on your Jira project board.
+
+## FAQ
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_jira_server.md:172]: Same corrections to this as to the JIRA Cloud page.

File: docs/4.3/enterprise/workflow/ssh_approval_mattermost.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+In Mattermost, go to System Console -> Integrations -> Enable Bot Account Creation -> Set to True.
+This will allow us to create a new bot account that the Teleport bot will use.
+
+Go back to your team, then Integrations -> Bot Accounts -> Add Bot Account.
+
+The new bot account will need Post All permission. 
+
+The confirmation screen after you've created the bot will give you the access token.
+ We'll use this in the config later.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_mattermost.md:36]: Space at beginning of line

File: docs/4.3/enterprise/workflow/ssh_approval_mattermost.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+#### Create an access-plugin role and user within Teleport 
+First off, using an existing Teleport Cluster, we are going to create a new Teleport
+ User and Role to access Teleport.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_mattermost.md:40]: Space at beginning of line

File: docs/4.3/enterprise/workflow/ssh_approval_mattermost.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+#### Create User and Role for access. 
+Log into Teleport Authentication Server, this is where you normally run `tctl`. Don't
+ change the username and the role name, it should be `access-plugin` for the plugin 
+ to work correctly.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_mattermost.md:45]: Space at beginning of line

File: docs/4.3/enterprise/workflow/ssh_approval_mattermost.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    logins: ['access-plugin']
+version: v3
+EOF
+
+# ...
+$ tctl create -f rscs.yaml
+```
+
+#### Export access-plugin Certificate
+Teleport Plugin uses the `access-plugin`role and user to peform the approval. We
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_mattermost.md:75]: `access-plugin` role and user (spacing)

`peform` -> `perform`

File: docs/4.3/enterprise/workflow/ssh_approval_mattermost.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+export the identify files, using [`tctl auth sign`](https://gravitational.com/teleport/docs/cli-docs/#tctl-auth-sign).
+
+```bash
+$ tctl auth sign --format=tls --user=access-plugin --out=auth --ttl=8760h
+# ...
+```
+
+The above sequence should result in three PEM encoded files being generated: auth.crt,
+ auth.key, and auth.cas (certificate, private key, and CA certs respectively).  We'll
+  reference these later in the bot config, and move them to an appropriate directory.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_mattermost.md:85]: Space at beginning of line

File: docs/4.3/enterprise/workflow/ssh_approval_mattermost.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+### Building from source
+
+```bash
+
+# Checkout teleport-plugins
+git clone git@github.com:gravitational/teleport-plugins.git
+cd teleport-plugins
+
+cd access/mattermost
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_mattermost.md:111]: Combine these `cd` commands into `cd teleport-plugins/access/mattermost`

File: docs/4.3/enterprise/workflow/ssh_approval_mattermost.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+make
+```
+
+### Configuring Mattermost bot
+
+Mattermost Bot uses a config file in TOML format. Generate a boilerplate config by 
+running the following command: 
+
+```
+teleport-mattermost configure > /etc/teleport-mattermost.yml
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_mattermost.md:121]: I think the path is `/etc/teleport-mattermost.toml` if it's a TOML file.

Also you likely won't be able to write to `/etc/teleport` if you're not `root`, so better to write the file to the local directory and then copy it.

```bash
$ teleport-mattermost configure > teleport-mattermost.toml
$ sudo mv teleport-mattermost.toml /etc
```

File: docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+This guide assumes you have
+
+* Teleport Enterprise 4.2.8 or newer with admin permissions and access to `tctl`
+* Pagerduty account already set, with access to creating a new API token. 
+
+### Create an access-plugin role and user within Teleport 
+First off, using an existing Teleport Cluster, we are going to create a new Teleport User and Role to access Teleport.
+
+#### Create User and Role for access. 
+Log into Teleport Authent Server, this is where you normally run `tctl`. Don't change the username and the role name, it should be `access-plugin` for the plugin to work correctly.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md:15]: `the Teleport Auth server`

File: docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+    logins: ['access-plugin']
+version: v3
+EOF
+
+# ...
+$ tctl create -f rscs.yaml
+```
+
+#### Export access-plugin Certificate
+Teleport Plugin uses the `access-plugin`role and user to perform the approval. We export the identify files, using [`tctl auth sign`](https://gravitational.com/teleport/docs/cli-docs/#tctl-auth-sign).
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md:47]: `access-plugin` role and user (spacing)

File: docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+
+# Check out the repo
+git clone git@github.com:gravitational/teleport-plugins.git
+cd teleport-plugins
+
+# Build the bot
+make access-pagerduty
+
+# Configure the plugin
+./access/pagerduty/build/teleport-pagerduty configure > teleport-pagertudy.toml
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md:76]: `pagertudy` -> `pagerduty`

File: docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+# Run the plugin, assuming you have teleport running: 
+./build/teleport-pagerduty start
+```
+The teleport-pagerduty executable should be placed onto a server that can access the auth server address.
+
+### Config file
+Teleport Pagerduty plugin has its own configuration file in TOML format. Before starting the plugin for the first time, you'll need to generate and edit that config file. 
+
+```bash
+teleport-pagerduty configure > /etc/teleport-pagerduty.toml
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md:87]: Same thing with not being able to write directly to `/etc/`

File: docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+```
+
+#### Editing the config file
+Afger generating the config, edit it as follows:
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md:91]: `Afger` -> `After`

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 14]
```diff
@@ -0,0 +1,223 @@
+## Teleport Plugins Setup Quickstart
+
+If you're using Slack, you can be notified of [new teleport permission requests](https://gravitational.com/teleport/docs/cli-docs/#tctl-request-ls), approve or deny them on Slack with Teleport Slack Plugin. This guide covers it's setup.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:3]: `new teleport permission requests` -> `new Teleport permission requests`

`covers it's setup` -> `covers its setup`

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+Note: The Approval Workflow only works with Pro and Enterprise version of Teleport.
+
+## Prerequisites
+- An Enterprise or Pro Teleport Cluster
+- Admin Privileges. With access and control of [`tctl`](https://gravitational.com/teleport/docs/cli-docs/#tctl)
+- Slack Admin Privileges to create an app and install it to your workspace.
+
+### Create an access-plugin role and user within Teleport 
+First off, using an existing
+Teleport Cluster, we are going to create a new Teleport User and Role to access Teleport.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:16]: Line spacing

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+    logins: ['access-plugin']
+version: v3
+EOF
+
+# ...
+$ tctl create -f rscs.yaml
+```
+
+#### Export access-plugin Certificate
+Teleport Plugin uses the `access-plugin`role and user to perform the approval. We export the identify files, using [`tctl auth sign`](https://gravitational.com/teleport/docs/cli-docs/#tctl-auth-sign).
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:49]: `access-plugin` role and user (spacing)

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 17]
```diff
@@ -0,0 +1,223 @@
...
+4. Obtain OAuth token and callback signing secret for the Teleport plugin config. 
+
+#### Creating the app
+
+https://api.slack.com/apps 
+
+App Name: Teleport
+Development Slack Workspace: Pick the workspace you'd like the requests to show up in.
+
+![Create Slack App](https://p197.p4.n0.cdn.getcloudapp.com/items/llu4EL7e/Image+2020-01-09+at+10.40.39+AM.png?v=d9750e4fdc77901e0c2ffb2dc6040aee)
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:77]: We should host this image locally

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 18]
```diff
@@ -0,0 +1,223 @@
...
+
+#### Setup Interactive Components
+
+When someone tries approving / denying a request on Slack and clicks an Approve / Deny button, Slack will send a request to the Teleport Slack Plugin. In Slack App's settings, you must provide the URL for Slack to use for this. 
+
+This URL must match the URL setting in Teleport Plugin settings file (we'll cover that later), and be publicly accessible.
+
+For now, just think of the URL you'll use and set it in the Slack App's settings screen in Features > Interactive Components > Request URL.
+
+![Interactive Componets](https://p197.p4.n0.cdn.getcloudapp.com/items/v1umg7J0/Image+2020-01-09+at+10.48.08+AM.png?v=5e231f17304f506db13c3ccf445b6682)
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:87]: Locally host the image

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 19]
```diff
@@ -0,0 +1,223 @@
...
+
+#### Selecting OAuth Scopes
+On the App screen, go to “OAuth and Permissions” under Features in the sidebar menu. Then scroll to Scopes, and add `chat.write` scope so that our plugin can post messages to your Slack channels.
+
+#### Add to Workspace
+
+![OAuth Token](https://p197.p4.n0.cdn.getcloudapp.com/items/E0uEg1ol/Image+2020-01-09+at+11.00.23+AM.png?v=1e28ff5bc4f7e0754acc9c7823f354a3)
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:94]: Locally host the image

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 20]
```diff
@@ -0,0 +1,223 @@
...
+
+## Installing the Teleport Slack Plugin
+To start using Teleport Plugins, you will need the teleport-slackbot executable.  See the [README](README.md) for building the teleport-slackbot executable in the Setup section.  Place the executable in the appropriate /usr/bin or /usr/local/bin on the server installation.
+
+### Configuration File
+Teleport Slack plugin has its own config file in TOML format. Before starting the plugin, you'll need to generate (or just copy the one below) and edit that config.
+
+To generate a config file, you can do this: 
+```bash
+teleport-slackbot configure > /etc/teleport-slackbot.toml
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:113]: Can't write directly to `/etc/`

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 21]
```diff
@@ -0,0 +1,223 @@
...
+## Test Run 
+
+Assuming that Teleport is running, and you've created the Slack app, the plugin config, and provided all the certificates — you can now run the plugin and test the workflow!
+
+`teleport-slackbot start`
+
+If everything works fine, the log output should look like this:
+
+```bash
+vm0:~/slackbot sudo ./teleport=slackbot start
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:159]: We should be able to run these plugins without needing to be `root` for security

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 22]
```diff
@@ -0,0 +1,223 @@
...
+### TSH User Login and Request Admin Role. 
+
+You can also test the full workflow from the user's perspective using `tsh`: 
+
+```bash
+➜ tsh login --request-roles=REQUESTED_ROLE
+Seeking request approval... (id: 8f77d2d1-2bbf-4031-a300-58926237a807)
+```
+
+You should now see a new request in Teleport, and a message about the request on Slack. You can approve or deny it and `tsh`should login successfully or error out right after you click an action button on Slack.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:199]: `tsh` should (spacing)

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md [hunk 23]
```diff
@@ -0,0 +1,223 @@
...
+[Service]
+Type=simple
+Restart=on-failure
+ExecStart=/usr/local/bin/teleport-slackbot start --config=/etc/teleport-slackbot.toml --pid-file=/var/run/teleport-slackbot.pid
+ExecReload=/bin/kill -HUP $MAINPID
+PIDFile=/var/run/teleport-slackbot.pid
+
+[Install]
+WantedBy=multi-user.target
+```
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_slack.md:219]: We should provide these `systemd` units for every plugin.

**Review by webvictim — COMMENTED** [review]
Sorry, couldn't resist a review...

File: docs/4.3/enterprise/workflow/index.md [hunk 24]
```diff
@@ -0,0 +1,132 @@
...
+
+```yaml
+kind: role
+metadata:
+  name: dba
+spec:
+  options:
+    # ...
+    # Only allows the contractor to use this role for 1hr from time of request.
+    options.max_session_ttl: 1hr
```
[the remark this answers is not in the document]
Comment (benarent) [reply docs/4.3/enterprise/workflow/index.md:51]: Good catch, I've fixed this.

**Comment by stevenGravy** [comment]
I made some minor edits.  One bit to just come back to is the location of the plugins.  I believe this is already on the radar but just mentioning again.

File: docs/4.3/enterprise/workflow/ssh_approval_jira_server.md [hunk 25]
```diff
@@ -1,12 +1,12 @@
 # Teleport Jira Plugin Setup
 
-If you're using Jira Cloud or Jira Server to manage your projects, you can also use it to monitor, approve, deny, or discuss Teleport permission requests. This quickstart will walk you through the setup.
+This guide will talk through how to setup Teleport with Pagerduty.   Teleport ↔ Pagerduty integration  allows you to treat Teleport access and permission requests as Pagerduty incidents — notifying the appropriate team, and approve or deny the requests via Pagerduty special action.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_jira_server.md:3]: Spacing is weird here, lots of double/triple spaces. Also the `↔` character might not exist in some fonts.

File: docs/4.3/enterprise/workflow/ssh_approval_jira_server.md [hunk 26]
```diff
@@ -1,12 +1,12 @@
...
 
-For the purpose of this quickstart, we assume you've already setup an [Enterprise Teleport Cluster](https://gravitational.com/teleport/docs/enterprise/quickstart-enterprise/). 
+!!! warning
+    The Approval Workflow only works with Teleport Enterprise as it's requires several roles.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_jira_server.md:6]: `as it requires`

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md (hunk 20 above)
Comment (benarent) [reply docs/4.3/enterprise/workflow/ssh_approval_slack.md]: Yeah, I found that out https://github.com/gravitational/teleport-plugins/issues/90

File: docs/4.3/enterprise/workflow/ssh_approval_mattermost.md (hunk 8 above)
Comment (benarent) [reply docs/4.3/enterprise/workflow/ssh_approval_mattermost.md]: Thanks Gus, I've updated the rest of the examples.

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md (hunk 16 above)
Comment (benarent) [reply docs/4.3/enterprise/workflow/ssh_approval_slack.md]: I've rewritten these.

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md (hunk 18 above)
Comment (benarent) [reply docs/4.3/enterprise/workflow/ssh_approval_slack.md]: Fixed.

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md (hunk 23 above)
Comment (benarent) [reply docs/4.3/enterprise/workflow/ssh_approval_slack.md]: I've added these

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md (hunk 14 above)
Comment (benarent) [reply docs/4.3/enterprise/workflow/ssh_approval_slack.md]: I've rewritten these.

File: docs/4.3/enterprise/workflow/ssh_approval_jira_server.md [hunk 27]
```diff
@@ -49,14 +49,15 @@ $ tctl create -f rscs.yaml
...
+```bash
 $ tctl auth sign --format=tls --user=access-plugin --out=auth --ttl=8760h
 # ...
 ```
 
 The above sequence should result in three PEM encoded files being generated: auth.crt, auth.key, and auth.cas (certificate, private key, and CA certs respectively).  We'll reference these later when [configuring Teleport-Plugins](#configuration-file).
 
-_Note: by default, tctl auth sign produces certificates with a relatively short lifetime. For production deployments, the --ttl flag can be used to ensure a more practical certificate lifetime. --ttl=8760h exports a 1 year token_
+!!! note "Certificate Lifetime"
+     By default, tctl auth sign produces certificates with a relatively short lifetime. For production deployments, the `--ttl` flag can be used to ensure a more practical certificate lifetime. `--ttl=8760h` exports a 1 year token
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_jira_server.md:60]: tctl auth sign -> `tctl auth sign`

File: docs/4.3/enterprise/workflow/ssh_approval_jira_server.md [hunk 28]
```diff
@@ -104,7 +105,18 @@ In the webhook settings page, make sure that the webhook will only send Issue Up
 
 ## Installing
 
-To start using Teleport Plugins, you will need the teleport-jirabot executable.  See the [README](README.md) for building the teleport-jiraport executable in the Setup section.  Place the executable in the appropriate /usr/bin or /usr/local/bin on the server installation.
+
+Where? .. Binary
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_jira_server.md]: ?

File: docs/4.3/enterprise/workflow/ssh_approval_mattermost.md [hunk 29]
```diff
@@ -2,8 +2,10 @@
 
 This guide will talk through how to setup Teleport with Mattermost.  Teleport ↔ Mattermost integration  allows teams to approve or deny Teleport access requests using [Mattermost](https://mattermost.com/) an open source messaging platform. 
 
-## Setup
+!!! warning
+    The Approval Workflow only works with Teleport Enterprise as it's requires several roles.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_mattermost.md:6]: `as it requires`

File: docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md [hunk 30]
```diff
@@ -1,11 +1,16 @@
 # Teleport Pagerduty Plugin Setup
 
-This guide will talk through how to setup Teleport with Mattermost.   Teleport ↔  Pagerduty integration  allows you to treat Teleport access and permission requests as Pagerduty incidents — notifying the appropriate team, and approve or deny the requests via Pagerduty special action.
+This guide will talk through how to setup Teleport with Pagerduty.   Teleport ↔ Pagerduty integration  allows you to treat Teleport access and permission requests as Pagerduty incidents — notifying the appropriate team, and approve or deny the requests via Pagerduty special action.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md:3]: weird spacing

File: docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md [hunk 31]
```diff
@@ -1,11 +1,16 @@
...
 
-## Prerequisites
-This guide assumes you have
+!!! warning
+    The Approval Workflow only works with Teleport Enterprise as it's requires several roles.
```
Comment (webvictim) [inline docs/4.3/enterprise/workflow/ssh_approval_pagerduty.md:6]: `as it requires`

**Review by webvictim — COMMENTED** [review]
quick review

File: docs/4.3/enterprise/workflow/ssh_approval_jira_server.md (hunk 28 above)
Comment (benarent) [reply docs/4.3/enterprise/workflow/ssh_approval_jira_server.md]: ha, it's updated now :-)

**Comment by webvictim** [comment]
Yeah, sorry, I got the email and was a bit overzealous :D

**Comment by benarent** [comment]
I seem to be running into a bunch of issues trying to get this into a clean state to merge into 4.3.

@awly The docs-whitespace isn't working for me, I'm getting a bunch ( 169 files, `.md-E`) when I run this. 

```
➜  teleport git:(docs/4.3/benarent/plugins) make docs-fix-whitespace
find: illegal option -- n
usage: find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
       find [-H | -L | -P] [-EXdsx] -f path [path ...] [expression]
find docs/ -type f -name '*.md' | xargs sed -i -E 's/\s+$//g'
```

**Comment by benarent** [comment]
Still a WIP but the diff should be much easier to review now. PTAL @webvictim There is also a bunch of duplicate content, I've tried to keep things consistent but not worried about minor difference as long as the instructions are clear.

File: docs/4.3/enterprise/workflow/ssh_approval_slack.md (hunk 21 above)
Comment (benarent) [reply docs/4.3/enterprise/workflow/ssh_approval_slack.md]: Created an issue here https://github.com/gravitational/teleport-plugins/issues/93

**Comment by webvictim** [comment]
> @awly The docs-whitespace isn't working for me, I'm getting a bunch ( 169 files, `.md-E`) when I run this.

It's because you're running on a Mac and GNU `find` is different to whatever MacOS uses.

**Comment by webvictim** [comment]
@benarent I committed a fix for the `Makefile` on Mac, `git pull` this branch again and it should work.

@awly Turned out to be this: https://github.com/gravitational/teleport/pull/3699/commits/7b20fa434c55e428cdc27585a7e7a79e8aa85c83

**Comment by benarent** [comment]
@webvictim odd, that is still giving me these extra -E files 
image

**Comment by webvictim** [comment]
That’s probably what https://github.com/gravitational/teleport/pull/3775 is intended to fix. That PR doesn’t have my fix in though (I committed to this branch rather than 4.3-base) which is why it’s still giving you the `find` error.

**Comment by webvictim** [comment]
@benarent I merged Andrew's fix PR and cherry-picked it here, do a `git pull` and try `make docs-fix-whitespace` again?

**Comment by benarent** [comment]
Merging this into the base fork, as to address meta build issues in one PR.

**Pull request merged**